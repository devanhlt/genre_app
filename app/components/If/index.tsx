import * as React from "react"

interface IfProps {
  condition?: boolean
  children: React.ReactNode
}

export const If = (props: IfProps) => {
  const { condition = false, children = <></> } = props

  if (condition) return children

  return null
}
