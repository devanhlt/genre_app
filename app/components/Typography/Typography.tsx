import { TxKeyPath, isRTL, translate } from "app/i18n"
import { appColors, typography } from "app/theme"
import i18n from "i18n-js"
import React from "react"
import { StyleProp, Text as RNText, TextProps as RNTextProps, TextStyle } from "react-native"

type Weights = keyof typeof typography.body02
type Presets = keyof typeof typography

export interface TypographyProps extends RNTextProps {
  /**
   * Text which is looked up via i18n.
   */
  tx?: TxKeyPath
  /**
   * The text to display if not using `tx` or nested components.
   */
  text?: string
  /**
   * Optional options to pass to i18n. Useful for interpolation
   * as well as explicitly setting locale or translation fallbacks.
   */
  txOptions?: i18n.TranslateOptions
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<TextStyle>
  /**
   * One of the different types of text presets.
   */
  preset?: Presets
  /**
   * Text weight modifier.
   */
  weight?: Weights

  /**
   * Color modifier.
   */
  color?: string
  /**
   * Children components.
   */
  children?: React.ReactNode
}

export function Typography(props: TypographyProps) {
  const {
    tx,
    txOptions,
    text,
    children,
    style: $styleOverride,
    color = appColors.common.characterPrimary,
    ...rest
  } = props

  const i18nText = tx && translate(tx, txOptions)
  const content = i18nText || text || children

  const preset: Presets = typography[props.preset] ? props.preset : "body02"

  const $styles = [$rtlStyle, { color }, typography[preset], $styleOverride]

  return (
    <RNText {...rest} style={$styles}>
      {content}
    </RNText>
  )
}

const $rtlStyle: TextStyle = isRTL ? { writingDirection: "rtl" } : {}
