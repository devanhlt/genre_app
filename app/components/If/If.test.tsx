import React from "react"

import renderer from "react-test-renderer"
import { If } from "./If"
import { Typography } from "../Typography"

describe("<If />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(
      <If condition={true}>
        <Typography text="Test" />
      </If>,
    )
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly when condition is true and children is empty", () => {
    const tree = renderer.create(<If condition={true}></If>)
    expect(tree).toMatchSnapshot()
  })

  it("should renders empty when condition is false", () => {
    const tree = renderer.create(
      <If condition={false}>
        <Typography text="Test" />
      </If>,
    )
    expect(tree).toMatchSnapshot()
  })

  it("should renders empty when condition is false and children is empty", () => {
    const tree = renderer.create(<If condition={false}></If>)
    expect(tree).toMatchSnapshot()
  })
  it("should renders empty when condition is undefined", () => {
    const tree = renderer.create(<If condition={undefined}></If>)
    expect(tree).toMatchSnapshot()
  })
})
