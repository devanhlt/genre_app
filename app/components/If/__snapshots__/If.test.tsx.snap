// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<If /> renders correctly 1`] = `
<Text
  style={
    [
      {},
      {
        "color": "#000F27",
      },
      {
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "400",
        "letterSpacing": 0.15,
        "lineHeight": 24,
      },
      undefined,
    ]
  }
>
  Test
</Text>
`;

exports[`<If /> renders correctly when condition is true and children is empty 1`] = `null`;

exports[`<If /> should renders empty when condition is false 1`] = `null`;

exports[`<If /> should renders empty when condition is false and children is empty 1`] = `null`;

exports[`<If /> should renders empty when condition is undefined 1`] = `null`;
